### PR TITLE
Port CA1813 (Avoid Unsealed Attributes) to System.Runtime.Analyzers.

### PR DESCRIPTION
**Pull request by srivatsn**
Change the fixer to use the DocumentEditor and remove the C#\VB specific fixers.

### CODE REVIEW

**Comment by srivatsn** [comment]
@shyamnamboodiripad @mavasani @heejaechang @tmeschter @jmarolf please take a look

**Comment by mavasani** [comment]
:+1:

File: src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Performance/AvoidUnsealedAttributes.Fixer.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Runtime.Analyzers
+{
+    /// <summary>
+    /// CA1813: Avoid unsealed attributes
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public class AvoidUnsealedAttributesFixer : CodeFixProvider
```
Comment (jaredpar) [inline src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Performance/AvoidUnsealedAttributes.Fixer.cs]: sealed?

File: src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/System.Runtime.Analyzers.Common.props [hunk 2]
```diff
@@ -12,6 +12,7 @@
       -Microsoft.Design#CA1014;
       -Microsoft.Design#CA1016;
       -Microsoft.Design#CA1017;
+      -Microsoft.Design#CA1018;
```
Comment (shyamnamboodiripad) [inline src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/System.Runtime.Analyzers.Common.props]: Shouldn't this be CA1813?

File: src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Performance/AvoidUnsealedAttributes.cs [hunk 3]
```diff
@@ -3,41 +3,54 @@
...
+                {
+                    AnalyzeSymbol((INamedTypeSymbol)context.Symbol, attributeType, context.ReportDiagnostic);
+                },
+                SymbolKind.NamedType);
+            });
         }
 
-        protected override void AnalyzeSymbol(INamedTypeSymbol namedType, Compilation compilation, Action<Diagnostic> addDiagnostic, AnalyzerOptions options, CancellationToken cancellationToken)
+        private static void AnalyzeSymbol(INamedTypeSymbol namedType, INamedTypeSymbol attributeType, Action<Diagnostic> addDiagnostic)
         {
```
Comment (shyamnamboodiripad) [inline src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Performance/AvoidUnsealedAttributes.cs]: [minor] Should we pass in cancellationToken and call cancellationToken.ThrowIfCancelled() or does driver already handle that?

File: src/Diagnostics/FxCop/Core/Microsoft.CodeAnalysis.FxCopAnalyzers.Common.props [hunk 4]
```diff
@@ -10,7 +10,6 @@
     <CodeAnalysisRuleSetOverrides>$(CodeAnalysisRuleSetOverrides);
       -Microsoft.Design#CA1008;
       -Microsoft.Design#CA1012;
-      -Microsoft.Design#CA1018;
```
Comment (shyamnamboodiripad) [inline src/Diagnostics/FxCop/Core/Microsoft.CodeAnalysis.FxCopAnalyzers.Common.props]: Did you remove the wrong rule? CA1813 exists in this list below.

**Comment by shyamnamboodiripad** [comment]
:+1:

File: src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Performance/AvoidUnsealedAttributes.cs (hunk 3 above)
Comment (srivatsn) [inline src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Performance/AvoidUnsealedAttributes.cs]: There's not much interesting work happening here to check for cancellation.

File: src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/System.Runtime.Analyzers.Common.props (hunk 2 above)
Comment (srivatsn) [inline src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/System.Runtime.Analyzers.Common.props]: It should also be CA1813. I'll move CA1813 in my next PR (which is moving one more rule).

**Pull request merged**